webapp/src/components/jira_ticket_tooltip/jira_ticket_tooltip.tsx

```diff
@@ -0,0 +1,312 @@
+            jiraTicketTitle = ticketDetails.summary;
+            jiraTicketAssigneeAvatarURI = ticketDetails.assigneeAvatar;
+            jiraTicketAssigneeName = ticketDetails.assigneeName;
+            jiraTicketStatusName = ticketDetails.statusKey;
+            jiraTicketDescription = ticketDetails.description;
+            jiraTicketVersions = ticketDetails.versions;
+            jiraTicketLabels = ticketDetails.labels;
+        }
+
+        return (
+            <div className='ticket-popover'>
+                <div className='popover-header'>
+                    <div className='popover-header__container'>
+                        <a
+                            href={jiraTicketURI}
+                            title='Go to ticket'
+                        >
+                            <img
+                                src={jiraAvatar}
+                                width={14}
+                                height={14}
+                                alt='jira-avatar'
+                                className='popover-header__avatar'
+                            />
+                        </a>
+                        <a
+                            href={jiraTicketURI}
+                            className='popover-header__keyword'
+                        >
+                            <span className='jira-ticket-key'>{jiraTicketKey}</span>
+                            <img
+                                alt='jira-issue-icon'
+                                width='14'
+                                height='14'
+                                src={jiraIssueIconURI}
+                            />
+                        </a>
+                    </div>
+                </div>
+                <div className='popover-body'>
+                    <div className='popover-body__title'>
+                        <a href={jiraTicketURI}>
+                            <h5>{this.truncateString(jiraTicketTitle as string, jiraTicketTitleMaxLength)}</h5>
+                        </a>
+                        {this.tagTicketStatus(jiraTicketStatusName as string)}
+                    </div>
+                    <div className='popover-body__description'>
+                        {jiraTicketDescription}
+                    </div>
+                    <div className='popover-body__labels'>
+                        {this.fixVersionLabel(jiraTicketVersions as string)}
+                        {this.labelList(jiraTicketLabels as string[])}
+                    </div>
+                </div>
+                <div className='popover-footer'>
+                    {jiraTicketAssigneeAvatarURI ? (
+                        <img
+                            className='popover-footer__assigner-profile'
+                            src={jiraTicketAssigneeAvatarURI}
+                            alt='jira assigner profile'
+                        />
+                    ) : (
+                        <span className='default-avatar'>
```

Comment:
Can we define this component in another file? Maybe create a generic `Avatar` component, that does this calculation of default avatar vs specific avatar